src/components/vertical-menu/responsive-vertical-menu/responsive-vertical-menu.component.tsx

```diff
@@ -0,0 +1,249 @@
+import React, {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+
+import { useResponsiveVerticalMenu } from "./responsive-vertical-menu.context";
+import {
+  StyledButton,
+  StyledGlobalVerticalMenuWrapper,
+  StyledResponsiveMenu,
+} from "./responsive-vertical-menu.style";
+
+import Box from "../../box";
+import Icon from "../../icon";
+import IconButton from "../../icon-button";
+import Modal from "../../modal";
+import useIsAboveBreakpoint from "../../../hooks/__internal__/useIsAboveBreakpoint";
+import useMediaQuery from "../../../hooks/useMediaQuery";
+
+import tagComponent, {
+  TagProps,
+} from "../../../__internal__/utils/helpers/tags";
+import { DepthProvider } from "./__internal__/depth.context";
+import { MenuFocusProvider } from "./__internal__/focus.context";
+
+export interface ResponsiveVerticalMenuProps extends TagProps {
+  /** The content of the menu */
+  children?: React.ReactNode;
+  /** The height of the primary and secondary menus */
+  height?: string;
+  /** The value (in pixels) at which the ResponsiveVerticalMenu will become responsive/modal */
+  responsiveBreakpoint?: number;
+  /** The width of the primary and secondary menus when in default mode */
+  width?: string;
+}
+
+const BaseMenu = ({
+  children,
+  height,
+  responsiveBreakpoint = 700,
+  width,
+  ...rest
+}: ResponsiveVerticalMenuProps) => {
+  const {
+    activeMenuItem,
+    buttonRef,
+    containerRef,
+    menuRef,
+    responsiveMode,
+    setActiveMenuItem,
+    setReducedMotion,
+    setResponsiveMode,
+  } = useResponsiveVerticalMenu();
+  const [active, setActive] = useState(false);
+  const largeScreen = useIsAboveBreakpoint(responsiveBreakpoint);
+  const [left, setLeft] = useState("auto");
+  const [responsiveWidth, setResponsiveWidth] = useState("100%");
+  const [top, setTop] = useState("auto");
+  const subMenuRef = useRef<HTMLDivElement>(null);
+  const reduceMotion = !useMediaQuery(
+    "screen and (prefers-reduced-motion: no-preference)",
+  );
+
+  useEffect(() => {
+    setReducedMotion?.(reduceMotion);
+    setResponsiveMode?.(!largeScreen || false);
+  }, [largeScreen, reduceMotion, setReducedMotion, setResponsiveMode]);
+
+  useLayoutEffect(() => {
```

Comment:
comment: it's typically bad practice to include refs in the dependency array as they're stable and won't cause re-renders etc. Maybe something like this would work?

```
  const menu = useRef(menuRef.current);
  const button = useRef(buttonRef.current);
  
  ...

  useLayoutEffect(() => {
    if (!menu?.current || !button?.current) return;

    if (active) {
      setResponsiveWidth(
        `${menu.current.getBoundingClientRect().width - 16}px`,
      );
    }

    if (activeMenuItem) {
      setLeft(`${menu.current.getBoundingClientRect().right}px`);
    } else {
      setLeft("auto");
    }

    if (activeMenuItem) {
      setTop(`${button.current.getBoundingClientRect().bottom}px`);
    } else {
      setTop("auto");
    }
  }, [active, activeMenuItem]);